### DISCUSSION TITLE
Unable to recompile the APK after making modifications

### DISCUSSION THREAD

**Discussion by Arohi02** [iBotPeaches/Apktool#3561] [Q&A]
I am facing error while rebuilding the apk file. So, I had made some modifications to the apk , like I had changed the file names , added and deleted some files. Now, when I am trying to rebuild it using the apktool I am facing an error.
I have tried using the following commands:-
-apktool b -f -d Surabhi_2.1.apk
-apktool b -f -d Surabhi_2.1 -o file1.apk 
-apktool b C:\Users\Desktop\apk2

and then it is giving the following error:-
```
W: C:\Users\Desktop\apk2\res\values\public.xml:1716: error: no definition for declared symbol 'com.DvaraEDairy.Surabhi:layout/abc_list_menu_item_radio'.
W: error: failed linking references.
brut.androlib.exceptions.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [C:\Users\AppData\Local\Temp\brut_util_Jar_15244054351977127111120208875318919289.tmp, link, -o, C:\Users\AppData\Local\Temp\APKTOOL13111307069270007342.tmp, --package-id, 127, --min-sdk-version, 19, --target-sdk-version, 28, --version-code, 1, --version-name, 1.0, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --allow-reserved-package-id, --no-compile-sdk-metadata, --warn-manifest-validation, -e, C:\Users\AppData\Local\Temp\APKTOOL6407037088237049661.tmp, -0, arsc, -I, C:\Users\AppData\Local\apktool\framework\1.apk, --manifest, C:\Users\Desktop\apk2\AndroidManifest.xml, C:\Users\Desktop\apk2\build\resources.zip]
```
Any idea how I can fix this and recompile my APK with the modifications?

**Comment by iBotPeaches** [comment]
The error helps point out your mistakes no?

```
W: C:\Users\Desktop\apk2\res\values\public.xml:1716: error: no definition for declared symbol 'com.DvaraEDairy.Surabhi:layout/abc_list_menu_item_radio'.
```

> Any idea how I can fix this and recompile my APK with the modifications?

Yes - follow the error outputted and resolve it.

**Comment by balram8264** [comment]
Please anyone give solution
apktool b -o output.apk payload               
I: Using Apktool 2.9.3
I: Checking whether sources has changed...
I: Checking whether resources has changed...
I: Building resources...
brut.androlib.exceptions.AndrolibException: brut.common.BrutException: could not exec (exit code = 2): [/tmp/brut_util_Jar_7626257
064607130280120840051442810200.tmp, compile, --dir, /payload/res, --legacy, -o, /payload/build/resources.zip]
W: /tmp/brut_util_Jar_7626257064607130280120840051442810200.tmp: 4: Syntax error: Unterminated quoted string.

**Reply by iBotPeaches** [reply]
@balram8264 - Sorry this is becoming absolutely ridiculous. You've spammed my blog with 3 comments and now responding to unrelated issues. What part of this is confusing to you? You can't seem to understand my responses or follow directions when I've answered this already.

**Reply by semeer37** [reply]
I also have same error. If anyone has solution direct us, please

**Comment by balram8264** [comment]
Sorry sir, I follow your guidance but someone block me at GitHub, please do
the needful

**Comment by M9204** [comment]
Dd u solve it, if yes how ?

**Comment by balram8264** [comment]
Sorry sir still I not find solution for that error. Can you help me

**Comment by marshmello61** [comment]
Using Apktool 2.10.0 with 8 thread(s).
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether resources has changed...
I: Building resources...
brut.androlib.exceptions.AndrolibException: brut.common.BrutException: could not exec (exit code = 132): [/tmp/brut_util_Jar_1279006676126871227154506694552937848.tmp, compile, --dir, /home/spoof2/MelloSpoofHack_010324/res, --legacy, -o, /home/spoof2/MelloSpoofHack_010324/build/resources.zip]
